### PR TITLE
Feature/modules view

### PR DESCRIPTION
**Pull request by ShpetimV**
# Modules View
Added the modules view inside the settings page. Connected it with an api where mock data for modules is predefined. 
Modules are being displayed, forms has been implemented for future editing and adding of modules but currently it only displays the values on the console.

image

image

### CODE REVIEW

File: client/src/app/(main)/settings/layout.tsx [hunk 1]
```diff
@@ -1,33 +1,40 @@
...
 import Link from "next/link";
 import { ReactNode } from "react";
+import { usePathname } from "next/navigation";
 
 type SettingsLayoutProps = {
-  children: ReactNode;
+    children: ReactNode;
 };
 
 // TODO: implement links and navigation for settings
```
Comment (mvoemel) [inline client/src/app/(main)/settings/layout.tsx:11]: remove

File: client/src/app/(main)/settings/modules/page.tsx [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    DialogContent,
+    DialogDescription,
+    DialogHeader,
+    DialogTitle,
+    DialogFooter
+} from "@/components/ui/dialog";
+import { ModuleForms } from "@/components/settings/modules/moduleForms";
+import { Progress } from "@/components/ui/progress"
+
+// Define a type for the module structure
```
Comment (mvoemel) [inline client/src/app/(main)/settings/modules/page.tsx:32]: remove

File: client/src/components/settings/modules/moduleForms.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    FormMessage,
+} from "@/components/ui/forms";
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { Textarea } from "@/components/ui/textarea";
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { z } from "zod";
+
+// Define a schema for the module structure
```
Comment (mvoemel) [inline client/src/components/settings/modules/moduleForms.tsx:17]: remove

File: client/src/components/settings/modules/moduleForms.tsx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    moduleUnderstanding: z.number().int().min(0, "Understanding must be a positive number").max(10, "Understanding must be at most 10"),
+    moduleTime: z.number().int().min(0, "Time must be a positive number").max(10, "Time must be at most 10"),
+    moduleImportance: z.number().int().min(0, "Importance must be a positive number").max(10, "Importance must be at most 10"),
+})
+
+type ModuleFormsProps = {
+    defaultValues?: Partial<z.infer<typeof formsSchema>>;
+}
+
+// Define your forms
```
Comment (mvoemel) [inline client/src/components/settings/modules/moduleForms.tsx:31]: remive

**Review by mvoemel — APPROVED** [review]

**Review by kryezleo — APPROVED** [review]
looks good

**Pull request merged**